### PR TITLE
Add getGoogleMapsMapType function

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: https://github.com/CartoDB/cartodb.js/issues/1734

**NOTE**

Now you need to pass the map to the get function because of WAX:

```js
var map = new google.maps.Map(document.getElementById('map'), {
  zoom: 10,
  center: {lat: 41.850, lng: -87.650}
});

var mapType = client.getGoogleMapsMapType(map);

map.overlayMapTypes.push(mapType);
```

This could be optimized in the future when WAX is removed.

### CODE REVIEW

File: src/api/v4/native/google-maps-map-type.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/* global google */
+var _ = require('underscore');
+var Layer = require('../layer');
+var GMapsCartoDBLayerGroupView = require('../../../geo/gmaps/gmaps-cartodb-layer-group-view');
+
+/**
+ * This object is a custom Google Maps MapType to enable feature interactivity
```
Comment (IagoLast) [inline src/api/v4/native/google-maps-map-type.js]: nice comment :D

**Comment by alonsogarciapablo** [comment]
`client.getGoogleMapsMapType(map);` could probably check if `google` is defined.

**Comment by alonsogarciapablo** [comment]
What about the CARTO attribution? Not sure how Google Maps handles this.

**Comment by Jesus89** [comment]
@iago we could extract some logic, but for example leaflet goes with `this._map.getContainer().style.cursor = 'pointer';` and gmaps with `this._map.setOptions({ draggableCursor: 'pointer' });`. I think we could extract only the `_triggerLayerFeatureEvent` function.

@alonsogarciapablo I think it is a good idea to check `goggle` (and also `leaflet`) in the `getGoogleMapsMapType` and `getLeafletLayer` functions.

We need to check how attribution is done in gmaps :+1:

**Comment by Jesus89** [comment]
It is not possible to change the attribution. So we are done!

**Pull request merged**